benches/note_decryption.rs

```diff
@@ -115,24 +113,24 @@ fn bench_note_decryption(c: &mut Criterion) {
 
         let mut group = c.benchmark_group("batch-note-decryption");
 
-        for size in array::IntoIter::new([10, 50, 100]) {
+        for size in &[10, 50, 100] {
```

Comment:
Instead of fixing this lint, the plan is that once we migrate to MSRV 1.56, we will enable Rust 2021 edition, and be able to use the fixed move iterator:
```suggestion
        for size in [10, 50, 100] {
```
This would avoid all of the `*size` noise changes below.

Reply:
```
116 |         for size in [10, 50, 100] {
    |                     ^^^^^^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
```